src/renderers/shared/shared/shouldUpdateReactComponent.js

```diff
@@ -22,24 +25,27 @@
  * @return {boolean} True if the existing instance should be updated.
  * @protected
  */
-function shouldUpdateReactComponent(prevElement, nextElement) {
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
-  if (prevEmpty || nextEmpty) {
-    return prevEmpty === nextEmpty;
+function shouldUpdateReactComponent(
+  prevElement: ReactElement | string | number | null | false,
+  nextElement: ReactElement | string | number | null | false,
+): boolean {
+  if (prevElement === null || prevElement === false) {
+    return nextElement === null || nextElement === false;
+  }
+
+  if (nextElement === null || nextElement === false) {
```

Comment:
The code is duplicated for those two conditions but they do not end up being computed twice because that's under two different branches